packages/fileeditor/src/widget.ts

```diff
@@ -204,6 +204,37 @@ export namespace FileEditor {
   };
 }
 
+/**
+ * A document widget for file editor widgets.
+ */
+export class FileEditorWidget extends DocumentWidget<FileEditor> {
+  /**
+   * Set URI fragment identifier for text files
+   */
+  setFragment(fragment: string): void {
+    const parseFragments = fragment.split('=');
+
+    // TODO: expand to allow more schemes of Fragment Identification Syntax
+    // reference: https://datatracker.ietf.org/doc/html/rfc5147#section-3
+    if (parseFragments[0] !== '#line') {
+      return;
+    }
+
+    // A range of lines can be provided, only respect first for now.
+    let firstLine = parseFragments[1].split(',')[0];
```

Comment:
Reading the norm, we should distinguish the single position and the range case. Because in the range case, the examples state:

>   http://example.com/text.txt#line=10,20
>
>   This URI identifies lines 11 to 20 of the text.txt MIME entity.  If
>   the MIME entity has fewer than 11 lines, it identifies the position
>   after the last line.  If the MIME entity has less than 20 but at
>   least 11 lines, it identifies the range from line 11 to the last line
>   of the MIME entity.
>
>   https://example.com/text.txt#line=,1
>
>   This URI identifies the first line.

So in case of a range the first line should be incremented by one and if undefined 0 should be taken.

Reply:
Thank you for catching it. I addressed the undefined range start in 439d735623fc5674be0a7ee5579057e734f48731 and  switched to 0-based positions in 9723fdf45aee993cc4908c8e3b7557db16e1b1c2. After checking the spec I  think there is no difference between the single position vs range, both are 0-based.